homeassistant/components/ecobee/weather.py

```diff
@@ -154,45 +160,63 @@ def wind_bearing(self):
-                }
-                if day["tempHigh"] == ECOBEE_STATE_UNKNOWN:
-                    break
-                if day["tempLow"] != ECOBEE_STATE_UNKNOWN:
-                    forecast[ATTR_FORECAST_TEMP_LOW] = float(day["tempLow"]) / 10
-                if day["pressure"] != ECOBEE_STATE_UNKNOWN:
-                    forecast[ATTR_FORECAST_PRESSURE] = int(day["pressure"])
-                if day["windSpeed"] != ECOBEE_STATE_UNKNOWN:
-                    forecast[ATTR_FORECAST_WIND_SPEED] = int(day["windSpeed"])
-                if day["visibility"] != ECOBEE_STATE_UNKNOWN:
-                    forecast[ATTR_FORECAST_WIND_SPEED] = int(day["visibility"])
-                if day["relativeHumidity"] != ECOBEE_STATE_UNKNOWN:
-                    forecast[ATTR_FORECAST_HUMIDITY] = int(day["relativeHumidity"])
-                forecasts.append(forecast)
-            return forecasts
-        except (ValueError, IndexError, KeyError):
+        if "forecasts" not in self.weather:
             return None
 
+        forecasts = list()
+        for day in range(1, 5):
+            fc = self._process_forecast(self.weather["forecasts"][day])
+            if fc is None:
+                continue
+            forecasts.append(fc)
+
+        if forecasts:
+            return forecasts
+        return None
+
     async def async_update(self):
         """Get the latest weather data."""
         await self.data.update()
         thermostat = self.data.ecobee.get_thermostat(self._index)
         self.weather = thermostat.get("weather", None)
+
+    def _process_forecast(self, json):
+        """Process a single ecobee API forecast to return expected values."""
+        try:
+            forecast = dict()
+
+            forecast[ATTR_FORECAST_TIME] = datetime.strptime(
+                json["dateTime"], "%Y-%m-%d %H:%M:%S"
+            ).isoformat()
+            forecast[ATTR_FORECAST_CONDITION] = ECOBEE_WEATHER_SYMBOL_TO_HASS[
+                json["weatherSymbol"]
+            ]
+            if json["tempHigh"] != ECOBEE_STATE_UNKNOWN:
+                forecast[ATTR_FORECAST_TEMP] = float(json["tempHigh"]) / 10
+            if json["tempLow"] != ECOBEE_STATE_UNKNOWN:
+                forecast[ATTR_FORECAST_TEMP_LOW] = float(json["tempLow"]) / 10
+            if json["windBearing"] != ECOBEE_STATE_UNKNOWN:
+                forecast[ATTR_FORECAST_WIND_BEARING] = int(json["windBearing"])
+            if json["windSpeed"] != ECOBEE_STATE_UNKNOWN:
+                forecast[ATTR_FORECAST_WIND_SPEED] = int(json["windSpeed"])
+
+            if json["pressure"] != ECOBEE_STATE_UNKNOWN:
+                forecast[ATTR_FORECAST_PRESSURE] = int(json["pressure"])
+            if json["visibility"] != ECOBEE_STATE_UNKNOWN:
+                forecast[ATTR_FORECAST_VISIBILITY] = int(json["visibility"]) / 1000
+            if json["relativeHumidity"] != ECOBEE_STATE_UNKNOWN:
+                forecast[ATTR_FORECAST_HUMIDITY] = int(json["relativeHumidity"])
```

Comment:
These (pressure, visibility, and humidity) are added as custom attributes in the ecobee weather platform but I'm not sure if they would end up getting used anywhere by the weather component. Should they just be removed to clean up?

Reply:
If we have the info, we can keep it around. Someone might find it useful somewhere.

Reply:
Yeah let's drop it.